app/javascript/components/hocs/render-while-error.jsx

```diff
@@ -0,0 +1,17 @@
+import PropTypes from 'prop-types';
+import {
+  branch,
+  compose,
+  renderComponent,
+  setPropTypes,
+} from 'react-recompose';
+
+export const renderWhileError = errorComponent =>
+  compose(
+    setPropTypes({
+      error: PropTypes.bool.isRequired,
+    }),
+    branch(({ error }) => !!error, renderComponent(errorComponent)),
+  );
```

Comment:
Again, I think `react-compose` obscures more than it reveals here

```suggestion
export const renderWhileError = Component, ErrorComponent => {
  const ComponentWithError = (error, ...props) => (
    error
      ? <ErrorComponent />
      : <Component {...props} )/>
  );
  
  ComponentWithError.propTypes({
    ...Component.propTypes,
    error: PropTypes.bool.isRequired,
  }
   
  return ComponentWithError;
}
```

Reply:
Repeating a comment from above, I think it would be cleaner to `import { ErrorComponent }` here instead of requiring it to be passed in.